lib/components/app/responsive-webapp.js

```diff
@@ -31,31 +26,42 @@ class ResponsiveWebapp extends Component {
 
   /** Lifecycle methods **/
 
-  componentWillReceiveProps (nextProps) {
-    const {query} = this.props
-
-    if (!isEqual(query, nextProps.query)) {
-      this.props.formChanged(query, nextProps.query)
+  componentDidUpdate (prevProps) {
+    const { currentPosition, location, query, title } = this.props
+    document.title = title
+    const urlParams = getUrlParams()
+    const newSearchId = urlParams.ui_activeSearch
+    const replanningTrip = newSearchId && this.props.activeSearchId && newSearchId !== this.props.activeSearchId
+    if (!isEqual(prevProps.query, query) && !replanningTrip) {
+      // Trigger on form change action if previous query is different from
+      // current one AND . This will determine whether a search needs to be made, the
+      // mobile view needs updating, etc.
+      this.props.formChanged(prevProps.query, query)
     }
 
     // check if device position changed (typically only set once, on initial page load)
-    if (this.props.currentPosition !== nextProps.currentPosition) {
-      if (nextProps.currentPosition.error || !nextProps.currentPosition.coords) return
+    if (currentPosition !== prevProps.currentPosition) {
+      if (currentPosition.error || !currentPosition.coords) return
       const pt = {
-        lat: nextProps.currentPosition.coords.latitude,
-        lon: nextProps.currentPosition.coords.longitude
+        lat: currentPosition.coords.latitude,
+        lon: currentPosition.coords.longitude
       }
 
       // if in mobile mode and from field is not set, use current location as from and recenter map
       if (isMobile() && this.props.query.from === null) {
         this.props.setLocationToCurrent({ type: 'from' })
         this.props.setMapCenter(pt)
         if (this.props.initZoomOnLocate) {
-          this.props.setMapZoom({zoom: this.props.initZoomOnLocate})
+          this.props.setMapZoom({ zoom: this.props.initZoomOnLocate })
         }
       }
     }
-
+    // If the path changes (e.g., via a back button press) check whether the
+    // main content needs to switch between, for example, a viewer and a search.
+    if (!isEqual(location.pathname, prevProps.location.pathname)) {
+      // console.log('url changed to', location.pathname)
+      this.props.matchContentToUrl(location)
```

Comment:
This is causing a 2nd dispatch of the `findRoute` action when selecting a new route in the route viewer.

Reply:
Workaround idea here: https://github.com/opentripplanner/otp-react-redux/issues/64